includes/wp-admin/table/class-following.php

```diff
@@ -123,41 +123,28 @@ public function process_action() {
 					return;
 				}
 
-				$profile = \sanitize_text_field( \wp_unslash( $_REQUEST['activitypub-profile'] ) );
-				$profile = \trim( $profile, '@' );
-
-				if ( \filter_var( $profile, FILTER_VALIDATE_EMAIL ) ) {
-					$remote_actor = Webfinger::resolve( $profile );
-					if ( ! \is_wp_error( $remote_actor ) ) {
-						$original = $profile;
-						$profile  = object_to_uri( $remote_actor );
-					}
-				} elseif ( empty( \wp_parse_url( $profile, PHP_URL_SCHEME ) ) ) {
-					// Add scheme if missing.
-					$profile = \esc_url_raw( 'https://' . \ltrim( $profile, '/' ) );
-				}
-
-				// Check if user has blocked the account.
-				if ( \in_array( $profile, Moderation::get_user_blocks( $this->user_id )['actors'], true ) ) {
+				$original = \sanitize_text_field( \wp_unslash( $_REQUEST['activitypub-profile'] ) );
+				$profile  = Actors::normalize_identifier( $original );
```

Comment:
I would move the normalizer into the `is_actor_blocked` function. It reduces the preconditions for the URL you want to check and simplifies the code.

Reply:
How does it speak against calling it directly inside of the blocked check?

Reply:
If it was moved into `is_actor_blocked` we'd break the functionality that allows users to put in a host only and still follow the blog actor. Like `activitypub.blog`.

The normalized identifier is a service to both `is_actor_blocked()` and `follow()` and prevents a webfinger from having to be resolved twice.

Reply:
FYI: It will not be resolved twice, we cache the request. so yes, it might be a bit of overload, but might simplify the code. but I am fine with improving over time.